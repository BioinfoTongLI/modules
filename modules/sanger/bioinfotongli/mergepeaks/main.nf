process BIOINFOTONGLI_MERGEPEAKS {
    tag "$meta.id"
    label 'process_single'

    // conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'quay.io/cellgeni/tiled_spotiflow:0.5.4-1':
        'quay.io/cellgeni/tiled_spotiflow:0.5.4-1' }"

    input:
    tuple val(meta), val(ch_ind), path(csvs)

    output:
    tuple val(meta), path("${output_name}"), emit: merged_peaks
    path "versions.yml"           , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    output_name = "${prefix}_merged_peaks_ch_${ch_ind}.wkt"
    """
    merge_overlapping_peaks.py run \
        ${csvs} \
        -output_name ${output_name} \
        ${args} \

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        bioinfotongli: \$(merge_overlapping_peaks.py version)
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    def ch_ind = meta.ch_ind ?: "-1"
    output_name = "${prefix}_merged_peaks_ch_${ch_ind}.wkt"
    """
    touch ${output_name}

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        bioinfotongli: \$(merge_overlapping_peaks.py version)
    END_VERSIONS
    """
}
