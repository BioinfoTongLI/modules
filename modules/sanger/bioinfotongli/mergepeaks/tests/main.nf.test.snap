{
    "dummy - csv ": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test_merged_peaks_ch_-1.wkt:md5,25c0054be03fcb2cd27b2165996bac90"
                    ]
                ],
                "1": [
                    "versions.yml:md5,b0aff58ea633428f5d954176a405a183"
                ],
                "merged_peaks": [
                    [
                        {
                            "id": "test"
                        },
                        "test_merged_peaks_ch_-1.wkt:md5,25c0054be03fcb2cd27b2165996bac90"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,b0aff58ea633428f5d954176a405a183"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.04.4"
        },
        "timestamp": "2025-02-07T14:47:18.430883388"
    },
    "dummy - csv - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test_merged_peaks_ch_-1.wkt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,b0aff58ea633428f5d954176a405a183"
                ],
                "merged_peaks": [
                    [
                        {
                            "id": "test"
                        },
                        "test_merged_peaks_ch_-1.wkt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,b0aff58ea633428f5d954176a405a183"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.04.4"
        },
        "timestamp": "2025-02-07T14:47:26.773474229"
    }
}